flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasMinInfoGain.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.DoubleParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for shared param minInfoGain. */
+public interface HasMinInfoGain<T> extends WithParams<T> {
```

Comment:
Since this is only used by `BaseGBTParams`, would it be simpler to define this parameter there instead of creating `HasMinInfoGain`?

Reply:
Good question! 

Actually, if more tree algorithms are planned in the future, `HasMinInfoGain` is then a common parameter. Same as many other parameters in GBT algorithms.   

BTW, in SparkML, these parameters are put in `DecisionTreeParams`. Maybe we can define `DecisionTreeParams`?